### PR TITLE
Fixed Size Tree #78

### PR DESCRIPTION
**Pull request by zyfrank**

### CODE REVIEW

**Comment by zyfrank** [comment]
Caused by hard producable hash error

File: proofs/tree.go [hunk 1]
```diff
@@ -384,13 +408,36 @@ func getSaltsFromMessage(message proto.Message) (salts []*proofspb.Salt, err err
...
 // the tree immutable.
 func (doctree *DocumentTree) Generate() error {
 	if doctree.filled {
 		return errors.New("tree already filled")
 	}
 
+	if doctree.fixedNoOfLeafs != 0 {
+		emptyNoToBeAdded := doctree.fixedNoOfLeafs - uint(len(doctree.leaves))
+		if emptyNoToBeAdded > 0 {
+			hash, err := emptyNodeHash(doctree.hash)
```
Comment (lucasvo) [inline proofs/tree.go]: This will cause an issue when we merge #80 (PR #81) Adding a comment here so we don't forget this.

Comment (lucasvo) [reply proofs/tree.go]: @touhonoob Would you be able to address this? I just merged this pull request. It should use the leaf node hash function here

Comment (touhonoob) [reply proofs/tree.go]: @lucasvo I'll have a look

Comment (touhonoob) [reply proofs/tree.go]: fixed in https://github.com/centrifuge/precise-proofs/pull/81/commits/4700de7049e357fcb20dd0fdddba0677c0c530b3

**Pull request merged**